spec/emerald/compiler_spec.rb

```diff
@@ -17,4 +17,24 @@
       expect(compiled_lisp).to eq("")
     end
   end
+
+  context "when the contents of the file define a variable" do
+    it "parses the lisp and returns an assigned global variable" do
+      lisp_file_contents = "(def y (- 1 6))"
+      compiled_lisp = Emerald::Compiler.new(lisp_file_contents).compile
+      expect(compiled_lisp).to eq("y = 1.0 - 6.0")
+    end
+
+    it "parses the lisp and returns an assigned local variable" do
+      lisp_file_contents = "(let [x 1 y (+ 3 4)] x)"
```

Comment:
They're both lists, because I didn't think that `[]` should have a data structure of it's own. This means that `[]` is acceptable instead of `()` which isn't ideal. The alternative is to treat `[]` as atoms instead. That probably makes more sense actually.